### PR TITLE
docs for adding DFE

### PR DESCRIPTION
**Pull request by mufernando**
Filling out the section for adding a new DFE to the docs.

The QCing process is basically the same as the demographic model, so I wasn't sure if I should repeat that section...

### CODE REVIEW

File: docs/development.rst [hunk 1]
```diff
@@ -1122,7 +1122,117 @@ to him until directed).
 Adding a DFE model
 ******************
 
-TODO: WRITE ME
+`stdpopsim` supports distribution of fitness effects (DFE) models from the 
+literature. To add a new DFE model to the catalog, you will need to:
+
+    1. `Fork the repository and create a branch`_
```
Comment (mufernando) [inline docs/development.rst:1128]: I was linking the sections from the demographic model docs, but maybe shouldn't? It's kinda weird that it jumps all the way up to that section.

File: docs/development.rst [hunk 2]
```diff
@@ -1122,7 +1122,116 @@ to him until directed).
 Adding a DFE model
 ******************
 
-TODO: WRITE ME
+`stdpopsim` supports distribution of fitness effects (DFE) models from the
+literature. To add a new DFE model to the catalog, you will need to:
+
+    1. `Fork the repository and create a branch`_
```
Comment (nspope) [inline docs/development.rst:1128]:
```suggestion
    1. `Fork the repository and create a branch`_ (as for demographic models)
```
You could give some indication that the link will jump to a different section

File: docs/development.rst [hunk 3]
```diff
@@ -1122,7 +1122,116 @@ to him until directed).
...
+            description=description,
+            long_description=long_description,
+            mutation_types=[neutral, negative, positive],
+            proportions=[p_neutral, p_negative, p_positive],
+            citations=citations,
+        )
+
+
+    _species.add_dfe(_dfe_func_name())
+
```
Comment (nspope) [inline docs/development.rst:1218]: Maybe give a link to the MutationType/DFE classes in the API documentation; something along the lines of 'For details regarding how the attributes of `MutationType` and `DFE` classes are used to model selection, see the API documentation here and the DFE tutorial here`

File: docs/development.rst [hunk 4]
```diff
@@ -1122,7 +1122,116 @@ to him until directed).
 Adding a DFE model
 ******************
 
-TODO: WRITE ME
+`stdpopsim` supports distribution of fitness effects (DFE) models from the
```
Comment (nspope) [inline docs/development.rst:1125]: Could link to DFE part of tutorial here

File: docs/development.rst [hunk 5]
```diff
@@ -1122,7 +1122,116 @@ to him until directed).
...
+        )
+
+
+    _species.add_dfe(_dfe_func_name())
+
+---------------------------------------------------
+Open a QC issue
+---------------------------------------------------
+
+Open a new issue requesting for a quality control check on your newly implemented
```
Comment (nspope) [inline docs/development.rst:1223]:
```suggestion
Open a new issue requesting a quality control check on your newly implemented
```

File: docs/development.rst [hunk 6]
```diff
@@ -1122,7 +1122,116 @@ to him until directed).
...
+
+    1. **PR for new model:**
+
+    2. **Original paper:**
+
+    3. **Parameter values:**
+
+    4. **Potential issues:**
+
+    5. **QC'er requests:**
```
Comment (nspope) [inline docs/development.rst:1236]: Could add something along the lines of "The QC process for DFEs follows that of demographic models [link]."

**Review by nspope — APPROVED** [review]
LGTM! Just some suggestions about linking to other parts of the docs.

File: docs/development.rst [hunk 7]
```diff
@@ -1122,7 +1122,116 @@ to him until directed).
...
+                doi="FILL ME",
+                reasons={stdpopsim.CiteReason.DFE},
+            )
+        ]
+
+        # Mutation Types based on the published DFE model, for example:
+        neutral = stdpopsim.MutationType()  # Neutral mutation type by default
+        negative = stdpopsim.MutationType(
+            dominance_coeff=0.5,
+            distribution_type="l",  # fixed selection coefficient
```
Comment (nspope) [inline docs/development.rst:1193]: Should this be `f`? Or is this meant to be `ln` with two parameters?

File: docs/development.rst [hunk 8]
```diff
@@ -1122,7 +1122,116 @@ to him until directed).
...
+
+.. code-block:: python
+
+    def _dfe_func_name():
+        id = "FILL ME"
+        description = "FILL ME"
+        long_description = """
+        FILL ME
+        """
+        populations = [
```
Comment (nspope) [inline docs/development.rst:1177]: Why do we need `populations` defined?

**Comment by nspope** [comment]
... also a couple of questions wrt to the code block stub.

**Comment by mufernando** [comment]
I resolved all of your suggestions @nspope! Should be good to go, I think.

**Comment by nspope** [comment]
LGTM, thanks @mufernando !

**Pull request merged**